Implemented 9, 10 functionalities

benhines1119 opened the pull request:
-implemented listOutstandingOrders() functionality 
-implemented printTransactions() functionality
-added getAllOrders() to GroceryStore
-added setOrderFields() to DataTransfer
-added setTranasctionFields() to DataTranser
-added setItemFields() to DataTransfer
-added several fields, getters and setters to DataTransfer
-added memberId to Transaction, getter and setter

benhines1119 committed:
Implemented functions 9, 10

-implemented listOutstandingOrders() functionality

-implemented printTransactions() functionality

-added getAllOrders() to GroceryStore

-added setOrderFields() to DataTransfer

-added setTranasctionFields() to DataTranser

-added setItemFields() to DataTransfer

-added several fields, getters and setters to DataTransfer

-added memberId to Transaction, getter and setter

-filled out TransactionsList class

benhines1119 committed:
Implemented 9, 10 functionalities

-implemented listOutstandingOrders() functionality

-implemented printTransactions() functionality

-added getAllOrders() to GroceryStore

-added setOrderFields() to DataTransfer

-added setTranasctionFields() to DataTranser

-added setItemFields() to DataTransfer

-added several fields, getters and setters to DataTransfer

-added memberId to Transaction, getter and setter

carterclark commented on src/store/collections/TransactionsList.java:
```diff
@@ -1,5 +1,5 @@
-}
+public class TransactionsList{
+	
+}
```
Not sure if he will nick us on this but for coding format there should be a space right here

carterclark commented on src/ui/UserInterface.java:
```diff
@@ -545,10 +546,177 @@ public void getMemberInfo() {
+						break;
+					}
+					if(!counter.hasNext()) {
+						//If you've managed to get here, there's no matching member.
```
These comments are really useful

carterclark commented on src/ui/UserInterface.java:
```diff
@@ -545,10 +546,177 @@ public void getMemberInfo() {
+		
+		//Loop the opportunity to input member ID until valid member ID is found or the user exits.
+		do {
+			System.out.print("Input member ID: ");
```
I think we need to explicitly say that the user can enter "END" with each printout from the system.

Also I think there is a get string method already built in? Just to account for the user giving invalid inputs

benhines1119 replied:
There is, but Eclipse complained that it "wasn't sure" what I was calling .equals() on was a String or not. The toString() can be removed, but I always assume an IDE has a better handle on code safety than myself.

carterclark replied:
Oh no, I mean a get string method made by pavil in the UserInterface. Line 75 in the user interface class

benhines1119 replied:
Oh yeah didn't see that. I'll change how I handle input.

paveldanek reviewed (commented):
Hey Ben, for functionalities 9 and 10, there was already a method in Member class (getTransactions()) and OrdersList class (inside the GroceryStore - outstanding())...

benhines1119 commented:
> 
> 
> Hey Ben, for functionalities 9 and 10, there was already a method in Member class (getTransactions()) and OrdersList class (inside the GroceryStore - outstanding())...

Yeah, but I'd have to call it on a Member object in the UI, which would expose that Member object to the UI. I'm trying to avoid that by passing a Result holding member data to a method that gets member transactions, then returns what it finds. If I can avoid importing/using any of our back-end objects in the UI, it's probably for the best.

benhines1119 committed:
Functionalities for 9 and 10

-added notification about entering 'END' to exit functionality 9 at the beginning

-implemented use of getSting for user input

paveldanek commented:
Yes, you're right. I looked into it a little further and I realized that more would have to be done, in order to make it work. Your coat looks pretty good, I just wish that we could extract some of it and make it a helper method. In order to reuse code.

Sent from my Apple Watch.

> On Mar 20, 2021, at 3:08 PM, benhines1119 ***@***.***> wrote:
> 
> Hey Ben, for functionalities 9 and 10, there was already a method in Member class (getTransactions()) and OrdersList class (inside the GroceryStore - outstanding())...
> 
> Yeah, but I'd have to call it on a Member object in the UI, which would expose that Member object to the UI. I'm trying to avoid that by passing a Result holding member data to a method that gets member transactions, then returns what it finds. If I can avoid importing/using any of our back-end objects in the UI, it's probably for the best.
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or unsubscribe.
